cmake/IgnCreateDocs.cmake

```diff
@@ -47,6 +47,12 @@ function(ign_create_docs)
   set(oneValueArgs API_MAINPAGE_MD AUTOGENERATED_DOC TUTORIALS_MAINPAGE_MD)
   set(multiValueArgs "TAGFILES" "ADDITIONAL_INPUT_DIRS")
 
+  option(IGN_BUILD_DOCS "Build docs" ON)
```

Comment:
Any reason we shouldn't make it OFF by default? Also, we have `BUILD_TESTING`, why not just `BUILD_DOCS`?

Reply:
> Any reason we shouldn't make it OFF by default?

I was preserving the current behavior.  I would like to see `make docs` a target that could be included/excluded from `all`, but I think that's too much of a shift without a major bump?  Maybe other people feel differently?

> why not just BUILD_DOCS?

Sure, I'm not too attached to this bikeshed color :)

Reply:
> Any reason we shouldn't make it OFF by default?

I think that if we make it `OFF` by default, then we should probably update the documentation elsewhere to indicate that building documentation is disabled by default so that users don't face issues when following instructions like [these](https://github.com/ignitionrobotics/ign-gazebo#documentation) (in particular, step 3).

> Also, we have BUILD_TESTING, why not just BUILD_DOCS?

Agreed - I think it would make the most sense to either prefix both arguments with `IGN_`, or to take the `IGN_` prefix out of both arguments. One thing to think about is that `BUILD_TESTING` and `BUILD_DOCS` are pretty generic arguments, which could perhaps introduce a naming collision for users that are using Ignition in other projects - so, I think it would be preferable to prefix these arguments with `IGN_`.

Reply:
BUILD_TESTING is actually an built-in in CMake as well.

Reply:
Updated to be BUILD_DOCS